crashreports/internal/win32/CrashReportContext.cpp

```diff
@@ -0,0 +1,115 @@
+#include <locale>
+#include <codecvt>
+#include <sstream>
+#include "client/windows/handler/exception_handler.h"
+#include "CrashReportContext.h"
+
+namespace {
+	bool uploadReport(
+		const wchar_t* dump_path,
+		const wchar_t* minidump_id,
+		void* context,
+		EXCEPTION_POINTERS* exinfo,
+		MDRawAssertionInfo* assertion,
+		bool succeeded)
+	{
+		CrashReportContext* crashReportContext = reinterpret_cast<CrashReportContext*>(context);
```

Comment:
reinterpret_cast is the cast operator of last resort.

static_cast is more appropriate when a void* context argument, given back to you by a C library, must be down-cast to a real type.

That applies to the other uses of reinterpret_cast in this commit.

And this declaration might also read better with auto.  (Which you did use elsewhere.)

Reply:
But thank you for not using C style casts!